### PR TITLE
Document MatchTrue, MatchFalse and MatchCondition readiness checks

### PR DESCRIPTION
**Pull request by LCaparelli**
Companion for https://github.com/crossplane/crossplane/pull/4399

I noticed MatchCondition was not yet documented. If this was intentional or if you'd rather it in a separate PR just let me know :-)

### CODE REVIEW

File: content/master/concepts/compositions.md [hunk 1]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+      base:
+        # Removed for brevity
+      readinessChecks:
+        - type: MatchTrue
+          fieldPath: status.atProvider.manifest.status.ready
+```
+{{<hint "tip" >}}
+Checking {{<hover label="matchTrue" line="11">}}MatchTrue{{</hover>}} doesn't
+require a `match` field.
+{{< /hint >}}
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1198]: Do you mean the `matchCondition:` field?

File: content/master/concepts/compositions.md [hunk 2]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+        - type: MatchTrue
+          fieldPath: status.atProvider.manifest.status.ready
+```
+{{<hint "tip" >}}
+Checking {{<hover label="matchTrue" line="11">}}MatchTrue{{</hover>}} doesn't
+require a `match` field.
+{{< /hint >}} 
+
+Alternatively, `MatchFalse` can be used for fields that express readiness with
+negative polarity.
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1201]:
```suggestion
`MatchFalse` matches fields that express readiness with the value `false`.
```

Simplifying the wording

File: content/master/concepts/compositions.md [hunk 3]
```diff
@@ -1130,6 +1132,100 @@ spec:
         - type: None
 ```
 
+#### Match a Condition
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1135]:
```suggestion
#### Match a condition
```
Titles should use sentence case

File: content/master/concepts/compositions.md [hunk 4]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+{{<hover label="condition" line="11">}}Condition{{</hover>}} considers the composed
+resource to be ready when it finds the expected condition type, with the
+expected status for it in its `status.conditions`.
+
+For example, consider 
+{{<hover label="condition" line="7">}}my-resource{{</hover>}}, which will be marked
+as ready if there is a condition of type 
+{{<hover label="condition" line="13">}}MyType{{</hover>}} with a status of
+{{<hover label="condition" line="14">}}Success{{</hover>}}.
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1144]:
```suggestion
For example, consider 
{{<hover label="condition" line="7">}}my-resource{{</hover>}}, which is
ready if there is a condition of type 
{{<hover label="condition" line="13">}}MyType{{</hover>}} with a status of
{{<hover label="condition" line="14">}}Success{{</hover>}}.
```
Fixing future tense

File: content/master/concepts/compositions.md [hunk 5]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+      base:
+        # Removed for brevity
+      readinessChecks:
+        - type: MatchCondition
+          matchCondition:
+            type: MyType
+            status: Success
+```
+
+#### Match a Boolean
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1163]:
```suggestion
#### Match a boolean
```

File: content/master/concepts/compositions.md [hunk 6]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+field within that resource is `true`.
+
+`MatchFalse` considers the composed resource to be ready when the value of a 
+field within that resource is `false`.
+
+For example, consider 
+{{<hover label="matchTrue" line="7">}}my-resource{{</hover>}}, which will be marked
+as ready if 
+{{<hover label="matchTrue" line="12">}} status.atProvider.manifest.status.ready{{</hover>}}
+is {{<hover label="matchTrue" line="11">}}true{{</hover>}}.
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1179]:
```suggestion
For example, consider 
{{<hover label="matchTrue" line="7">}}my-resource{{</hover>}}, which is
ready if 
{{<hover label="matchTrue" line="12">}} status.atProvider.manifest.status.ready{{</hover>}}
is {{<hover label="matchTrue" line="11">}}true{{</hover>}}.
```
fixes future tense.

File: content/master/concepts/compositions.md [hunk 7]
```diff
@@ -1130,6 +1132,100 @@ spec:
...
+{{< /hint >}} 
+
+Alternatively, `MatchFalse` can be used for fields that express readiness with
+negative polarity.
+
+For example, consider 
+{{<hover label="matchFalse" line="7">}}my-resource{{</hover>}}, which will be marked
+as ready if 
+{{<hover label="matchFalse" line="12">}} status.atProvider.manifest.status.pending{{</hover>}}
+is {{<hover label="matchFalse" line="11">}}false{{</hover>}}.
```
Comment (plumbis) [inline content/master/concepts/compositions.md:1207]:
```suggestion
For example, consider 
{{<hover label="matchFalse" line="7">}}my-resource{{</hover>}}, is
ready if 
{{<hover label="matchFalse" line="12">}} status.atProvider.manifest.status.pending{{</hover>}}
is {{<hover label="matchFalse" line="11">}}false{{</hover>}}.
```
fixes future tense.

File: content/v1.13/concepts/compositions.md [hunk 8]
```diff
@@ -1130,6 +1131,34 @@ spec:
         - type: None
 ```
 
+#### Match a Condition
```
Comment (plumbis) [inline content/v1.13/concepts/compositions.md:1134]:
```suggestion
#### Match a condition
```

File: content/v1.13/concepts/compositions.md [hunk 9]
```diff
@@ -1130,6 +1131,34 @@ spec:
...
+{{<hover label="condition" line="11">}}Condition{{</hover>}} considers the composed
+resource to be ready when it finds the expected condition type, with the
+expected status for it in its `status.conditions`.
+
+For example, consider 
+{{<hover label="condition" line="7">}}my-resource{{</hover>}}, which will be marked
+as ready if there is a condition of type 
+{{<hover label="condition" line="13">}}MyType{{</hover>}} with a status of
+{{<hover label="condition" line="14">}}Success{{</hover>}}.
```
Comment (plumbis) [inline content/v1.13/concepts/compositions.md:1143]:
```suggestion
For example, consider 
{{<hover label="condition" line="7">}}my-resource{{</hover>}}, which is
ready if there is a condition of type 
{{<hover label="condition" line="13">}}MyType{{</hover>}} with a status of
{{<hover label="condition" line="14">}}Success{{</hover>}}.
```

**Comment by LCaparelli** [comment]
Ah thank you @plumbis, that's my bad, I forgot to run the checks locally :sweat_smile:

I'll commit fixes shortly.

**Comment by phisco** [comment]
@LCaparelli, gentle reminder 😅

File: content/master/concepts/compositions.md (hunk 1 above)
Comment (LCaparelli) [reply content/master/concepts/compositions.md:1198]: No, it's more like a `matchSomethingSomething` field. Eg, the `MatchInteger` type has a companion `mathInteger` type.

`MatchTrue` and `MatchFalse` don't require additional fields. Which I guess would be a better way to describe this:

```suggestion
{{<hint "tip" >}}
Checking {{<hover label="matchTrue" line="11">}}MatchTrue{{</hover>}} doesn't
require setting additional fields.
{{< /hint >}} 
```

Sounds good?

PS: I copied this section from [here](https://github.com/crossplane/docs/blob/969b94d2809ae3d826c9c02ed890925e8c0d14be/content/master/concepts/compositions.md?plain=1#L1077), I'm guessing it would also make sense to adjust that. Wdyt?

https://github.com/crossplane/docs/blob/969b94d2809ae3d826c9c02ed890925e8c0d14be/content/master/concepts/compositions.md?plain=1#L1106-L1109

**Comment by LCaparelli** [comment]
> @LCaparelli, gentle reminder 😅

Thanks hehe, just got back from vacation and this one escaped my final wrapping up process before disappearing :grin:

File: content/master/concepts/compositions.md (hunk 1 above)
Comment (LCaparelli) [reply content/master/concepts/compositions.md:1195]: Ah, apparently "additional" is a bit too wordy for Vale. :thinking: 

This seems ok:
```suggestion
{{<hint "tip" >}}
Checking {{<hover label="matchTrue" line="11">}}MatchTrue{{</hover>}} doesn't
require setting any other fields.
{{< /hint >}} 
```

Comment (LCaparelli) [reply content/master/concepts/compositions.md]: @plumbis just pushed a commit with this fix, lmkwyt :grinning:

**Review by plumbis — APPROVED** [review]
